### PR TITLE
test: integration testing and ci

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: .github/workflows/pull_request.yaml [hunk 1]
```diff
@@ -42,6 +41,33 @@ jobs:
...
+        ports:
+          - 5432:5432
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      minio:
+        image: bitnami/minio:latest
```
Comment (CptSchnitz) [inline .github/workflows/pull_request.yaml:65]: dont use latest

File: commitlint.config.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
     extends: ['@commitlint/config-conventional'],
     rules: {
-        "scope-enum": [2, "always", ["deps", "configurations"]],
+        "scope-enum": [2, "always", ["deps", "configurations", "ci"]],
```
Comment (CptSchnitz) [inline commitlint.config.js:4]: pretty sure "ci" is not needed here as its a type and not a scope.

File: src/app.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Logger } from '@map-colonies/js-logger';
+import { DependencyContainer } from 'tsyringe';
+import { JOB_QUEUE_PROVIDER, SERVICES } from './common/constants';
+import { timerify } from './common/util';
+import { JobQueueProvider } from './retiler/interfaces';
+import { TileProcessor } from './retiler/tileProcessor';
+import { TileWithMetadata } from './retiler/types';
+
+export async function runApp(container: DependencyContainer): Promise<void> {
```
Comment (CptSchnitz) [inline src/app.ts:9]: it would be nice to have a better name

File: src/index.ts [hunk 4]
```diff
@@ -19,23 +16,12 @@ void registerExternalValues()
   .then(async (container) => {
     depContainer = container;
 
-    const livenessProbe = initLivenessProbe(container);
+    initLivenessProbe(container);
 
-    const processor = container.resolve(TileProcessor);
-    const queueProv = container.resolve<JobQueueProvider>(JOB_QUEUE_PROVIDER);
-    const logger = container.resolve<Logger>(SERVICES.LOGGER);
+    await runApp(container);
```
Comment (CptSchnitz) [inline src/index.ts:21]: both of those should be in containerConfig as factories.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: .github/workflows/pull_request.yaml (hunk 1 above)
Comment (melancholiai) [reply .github/workflows/pull_request.yaml]: done.

File: commitlint.config.js (hunk 2 above)
Comment (melancholiai) [reply commitlint.config.js]: done.

File: src/app.ts (hunk 3 above)
Comment (melancholiai) [reply src/app.ts]: done.

File: src/index.ts (hunk 4 above)
Comment (melancholiai) [reply src/index.ts]: done.

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**